### ISSUE TITLE
addcmul on CUDA does not have the correct FMA behavior

### ISSUE THREAD

**Issue by Yujia-Yan** [pytorch/pytorch#89764]
### 🐛 Describe the bug

To reproduce the result:

``` python
import torch

x = torch.randn(1000)
y = torch.randn(1000)

result = x*y
diff = torch.addcmul(-result, x, y)
print("diff_cpu:", diff.abs().max().item())

x = x.cuda()
y = y.cuda()

result = x*y
diff = torch.addcmul(-result, x, y)
print("diff_cuda:", diff.abs().max().item())

# obtaining reference using double precision
xDouble = x.double()
yDouble = y.double()
diffDouble = xDouble*yDouble - result
print("diff_double:", diffDouble.abs().max().item())

```

Outputs:
``` 
diff_cpu: 2.250976649520453e-07
diff_cuda: 0.0
diff_double: 2.250976649520453e-07
```

### Versions

Pytorch installed via PIP

Outputs from collect_env.py:

Collecting environment information...
PyTorch version: 1.13.0+cu117
Is debug build: False
CUDA used to build PyTorch: 11.7
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.1 LTS (x86_64)
GCC version: (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0
Clang version: Could not collect
CMake version: version 3.22.1
Libc version: glibc-2.35

Python version: 3.10.6 (main, Nov  2 2022, 18:53:38) [GCC 11.3.0] (64-bit runtime)
Python platform: Linux-5.15.0-53-generic-x86_64-with-glibc2.35
Is CUDA available: True
CUDA runtime version: 11.8.89
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: 
GPU 0: NVIDIA GeForce RTX 3090
GPU 1: NVIDIA GeForce RTX 3090

Nvidia driver version: 520.61.05
cuDNN version: Probably one of the following:
/usr/local/cuda-9.1/targets/x86_64-linux/lib/libcudnn.so.6
/usr/local/cuda-9.1/targets/x86_64-linux/lib/libcudnn.so.7
/usr/local/lib/libcudnn.so.6
/usr/local/lib/libcudnn.so.7
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] memory-efficient-attention-pytorch==0.0.27
[pip3] numpy==1.23.4
[pip3] torch==1.13.0
[pip3] torch-tb-profiler==0.4.0
[pip3] torchaudio==0.13.0
[pip3] torchvision==0.14.0
[conda] Could not collect

**Comment by vulkomilev** [comment]
I will take a look

**Comment by nobody81127** [comment]
I applied a floating-point error analysis algorithm to calculate the error bounds for the result of addcmul, and I found that 0 falls within this range. Therefore, I believe this is indeed a floating-point error, likely caused by differences in the order of operations. Therefore, I suggest closing this issue. @Yujia-Yan @vulkomilev

**Comment by abs-hello** [comment]
Is there any update?

**Comment by Yujia-Yan** [comment]
> This is a floating-point error rather than a functional defect. It is likely caused by the addcmul operation calculating the result of x* y - result at high precision, where cumulative rounding errors are expected to occur.

No, computing rounding error is the expected  behavior of FMA in this example. fma for float fma(-x*y, x,y) should give the rounding error identical  using double for computing the rounding error x*y-result.
